packages/element-lib/src/sidetree-v2/protocol/resolve.js

```diff
@@ -113,11 +132,51 @@ const resolve = sidetree => async (did) => {
   // TODO test that
   // eslint-disable-next-line max-len
   operations.sort((op1, op2) => op1.transaction.transactionNumber - op2.transaction.transactionNumber);
-  // TODO operation validation
-  const didDocument = await operations
+  const createAndRecoverAndRevokeOperations = operations.filter((op) => {
+    const type = op.operation.decodedOperation.header.operation;
+    return ['create', 'recover', 'delete'].includes(type);
+  });
+  // Apply "full" operations first.
+  let lastValidFullOperation;
+  let didDocument = await createAndRecoverAndRevokeOperations
     .reduce((promise, operation) => {
-      return promise.then(acc => applyOperation(acc, operation.operation));
+      return promise.then(async (acc) => {
+        const { valid, newState } = await applyOperation(
+          acc, operation.operation, lastValidFullOperation,
+        );
+        if (valid) {
+          lastValidFullOperation = operation;
+        }
+        return newState;
+      });
     }, Promise.resolve(undefined));
+  // If no full operation found at all, the DID is not anchored.
+  if (lastValidFullOperation === undefined) {
```

Comment:
This is where we may want to use the initial-values stuff discussed in core

Reply:
Yes good point